### PR TITLE
Add support for deployment label

### PR DESCRIPTION
**Pull request by iansul**

### CODE REVIEW

File: Tasks/AzureCloudPowerShellDeployment/Publish-AzureCloudDeployment.ps1 [hunk 1]
```diff
@@ -178,6 +184,8 @@ Write-Host "StorageAccount= $StorageAccount"
 Write-Host "CsPkg= $CsPkg"
 Write-Host "CsCfg= $CsCfg"
 Write-Host "Slot= $Slot"
+Write-Host "DeploymentLabel= $DeploymentLabel"
+Write-Host "AppendDateTimeToLabel= $AAppendDateTimeToLabel"
```
Comment (jeffyoung) [inline Tasks/AzureCloudPowerShellDeployment/Publish-AzureCloudDeployment.ps1]: Looks like an extra A at the beginning of the variable being assigned.

File: Tasks/AzureCloudPowerShellDeployment/Publish-AzureCloudDeployment.ps1 [hunk 2]
```diff
@@ -202,27 +210,60 @@ if (!$azureService)
 
 $diagnosticExtensions = Get-DiagnosticsExtensions $StorageAccount $CsCfg
 
+$label = $DeploymentLabel
+
+$appendDateTime = Convert-String $AppendDateTimeToLabel Boolean
+
+if($label -and $appendDateTime)
```
Comment (jeffyoung) [inline Tasks/AzureCloudPowerShellDeployment/Publish-AzureCloudDeployment.ps1]: We like spaces between our if and opening parentheses.  :-)  there a couple others further down as well.

**Comment by jeffyoung** [comment]
Also, on line 16 of the JSON file, you'll want to bump up the version of the task so, when it gets deployed, all existing agents will get the update.

**Comment by iansul** [comment]
Thanks for the review Jeff

**Comment by jeffyoung** [comment]
I think the last thing is that there should have been an update to the loc.json file during your local gulp build.  That would need to get pushed too so our CI build can create the proper package.  WHen changing tasks, we (almost) always rev the version and, if adding properties, update the task json and push the loc.json as well.  I believe the gulp build will pend the generated files.  Have a look to see if it did (it should have).

**Comment by iansul** [comment]
Oh it did, for every task :), cause I just did 'gulp' at the commandline and it built every task.
 I wasn't sure if loc was done at official build time. I will push everything in the folder

**Comment by jeffyoung** [comment]
Thanks Ian.  Once this is merged, we should automatically get a CI build and a new package built.  Once we test that, we can get it into an upcoming deployment.

**Pull request merged**